graph-subscriptions-api/src/config.rs

```diff
@@ -15,133 +16,159 @@ pub struct Config {
-            }
-        };
-    let graph_subscription_logs_kafka_group_id =
-        match dotenv::var("GRAPH_SUBSCRIPTION_LOGS_KAFKA_GROUP_ID") {
-            Ok(group_id) => group_id,
-            Err(_) => {
-                panic!("GRAPH_SUBSCRIPTION_LOGS_KAFKA_GROUP_ID environment variable is required")
-            }
-        };
-    let graph_subscription_logs_kafka_topic_id =
-        match dotenv::var("GRAPH_SUBSCRIPTION_LOGS_KAFKA_TOPIC_ID") {
-            Ok(group_id) => group_id,
-            Err(_) => {
-                panic!("GRAPH_SUBSCRIPTION_LOGS_KAFKA_TOPIC_ID environment variable is required")
-            }
-        };
-    // attempt to parse and build the additional kafka config
-    let mut kafka_additional_config = BTreeMap::<String, String>::new();
-    if let Some(security_protocol) =
-        dotenv::var("GRAPH_SUBSCRIPTION_LOGS_KAFKA_SECURITY_PROTOCOL").ok()
-    {
-        kafka_additional_config.insert("security.protocol".to_string(), security_protocol);
-    }
-    if let Some(mechanism) = dotenv::var("GRAPH_SUBSCRIPTION_LOGS_KAFKA_SASL_MECHANISM").ok() {
-        kafka_additional_config.insert("sasl.mechanism".to_string(), mechanism);
-    }
-    if let Some(username) = dotenv::var("GRAPH_SUBSCRIPTION_LOGS_KAFKA_SASL_USERNAME").ok() {
-        kafka_additional_config.insert("sasl.username".to_string(), username);
-    }
-    if let Some(password) = dotenv::var("GRAPH_SUBSCRIPTION_LOGS_KAFKA_SASL_PASSWORD").ok() {
-        kafka_additional_config.insert("sasl.password".to_string(), password);
-    }
-    if let Some(ssl_ca_loc) = dotenv::var("GRAPH_SUBSCRIPTION_LOGS_KAFKA_SSL_CA_LOC").ok() {
-        kafka_additional_config.insert("ssl.ca.location".to_string(), ssl_ca_loc);
+impl KafkaConfig {
+    pub fn build(mut conf: KafkaConfig) -> BTreeMap<String, String> {
+        let mut settings = conf.0.clone();
+        settings.append(&mut conf.0);
+
+        settings
     }
-    if let Some(ssl_cert_loc) = dotenv::var("GRAPH_SUBSCRIPTION_LOGS_KAFKA_SSL_CERT_LOC").ok() {
-        kafka_additional_config.insert("ssl.certificate.location".to_string(), ssl_cert_loc);
+}
+
+impl Default for KafkaConfig {
+    fn default() -> Self {
+        let settings = [
+            ("bootstrap.servers", "PLAINTEXT://127.0.0.1:9092"),
+            ("group.id", "graph-gateway"),
+            ("message.timeout.ms", "3000"),
+            ("queue.buffering.max.ms", "1000"),
+            ("queue.buffering.max.messages", "100000"),
+            ("enable.partition.eof", "false"),
+            ("enable.auto.commit", "false"),
+        ];
+        Self(
+            settings
+                .into_iter()
+                .map(|(k, v)| (k.to_owned(), v.to_owned()))
+                .collect(),
+        )
     }
-    if let Some(ssl_key_loc) = dotenv::var("GRAPH_SUBSCRIPTION_LOGS_KAFKA_SSL_KEY_LOC").ok() {
-        kafka_additional_config.insert("ssl.key.location".to_string(), ssl_key_loc);
+}
+
+impl PartialEq for KafkaConfig {
```

Comment:
You should be able to automatically derive `PartialEq` for `KafkaConfig` unsing the `#[derive(...)]` syntax